himan-scripts/traffic_weather_warning.lua

```diff
@@ -0,0 +1,354 @@
+logger:Info("Alku: traffic warning calculation")
+
+function min_max_mean(tbl)
+
+  local min = tbl[1]
+  local max = tbl[1]
+  local sum = 0
+
+  for i = 1, #tbl do
+      if tbl[i] > max then
+          max = tbl[i]
+      end
+      if tbl[i] < min then
+          min = tbl[i]
+      end
+      sum = sum + tbl[i]
+  end
+  local mean = sum / #tbl
+  return tostring(min).. ' ' .. tostring(max).. ' '.. tostring(mean)
+end
+
+function warning_reason(warning, ice_warning, frost)
+  local reason = {}
+  for i=1, #warning do
+    reason[i] = 0 
+    reason[i] = warning[i] >= frost[i] and warning[i] or reason[i]
+    reason[i] = warning[i] >= ice_warning[i] and warning[i] or reason[i]
+    reason[i] = frost[i] >= warning[i] and frost[i] or reason[i]
+    reason[i] = frost[i] >= ice_warning[i] and frost[i] or reason[i]
+    reason[i] = ice_warning[i] >= warning[i] and ice_warning[i] or reason[i]
+    reason[i] = ice_warning[i] >= frost[i] and ice_warning[i] or reason[i]
+  end
+  return reason
+end
+
+
+function combine_warning_factors(warning, thawing, freezing, wind, radglo_w)
+  warning = {}
+  for i=1, #thawing do
+    warning[i] = thawing[i] * freezing[i] * wind[i] * radglo_w[i]
+  end
+  return warning
+end
+
+function frost_effect(nlm, tk, tdk, ff, radglo3)
+  local frost, c_f, t_f, tdk_f, w_f, r_f, tc, tdc = {}, {}, {}, {}, {}, {}, {}, {}
+
+  for i = 1, #nlm do
+    tc[i] = tk[i] - 273.15
+    tdc[i] = tdk[i] - 273.15
+
+    c_f[i] = (125 - nlm[i])/100
+
+    t_f[i] = (-0.25)*tc[i] + 1.25 < 0 and 0 or (-0.25)*tc[i] + 1.25
+    t_f[i] = t_f[i] > 1 and 1 or t_f[i]
+    
+    tdk_f[i] = (-0.5) * (tc[i] - tdc[i]) + 2 < 0 and 0 or (-0.5) * (tc[i] - tdc[i]) + 2
+    tdk_f[i] = tdk_f[i] > 2 and 2 or tdk_f[i]
+
+    w_f[i] = 0.125 * ff[i] + 1 < 0.5 and 0.5 or 0.125 * ff[i] + 1
+    w_f[i] = w_f[i] > 2 and 2 or w_f[i]
+
+    r_f[i] = (-0.01) * radglo3[i] + 1 < 0 and 0 or (-0.01) * radglo3[i] + 1
+    r_f[i] = r_f[i] > 1 and 1 or r_f[i]
+
+    frost[i] = (r_f[i] * c_f[i] * t_f[i] * tdk_f[i] * w_f[i]) - 1
+    frost[i] = frost[i] < 0 and 0 or frost[i]
+    frost[i] = frost[i] > 2 and 2 or frost[i]
+
+  end
+  return frost
+end
+ 
+  
+function freezing_rain_effect(rr, prec, tk)
+  local freezing_rr_effect = {}
+  for i = 1, #rr do
+    if prec[i] == 4 or prec[i] == 5 then
```

Comment:
Tästä jäin miettimään että mitä smarttool palauttaa silloin jos ehto ei täyty.

Funktiokutsu on siis tämä:
`time_sum(rr_ec, -1, 1, "par57_ec = 4 || par57_ec =5")`

Ja jos on vaikka lumisadetta, niin palauttaako tämä arvon 0 eli ei sadetta, vai arvon MISS eli puuttuvaa. Jos ensimmäinen, niin tämä lua-versio toimii oikein, jos jälkimmäinen niin siitä seuraisi että koko freezing rain effect olisi puuttuva.